main/src/ca/uwaterloo/flix/api/lsp/provider/completion/Completion.scala

```diff
@@ -316,11 +334,35 @@ object Completion {
   /**
     * Represents a keyword completion.
     *
-    * @param name the name of the keyword.
-    * @param priority the completion priority of the keyword.
+    * @param name      the name of the keyword.
+    * @param priority  the completion priority of the keyword.
     */
   case class KeywordCompletion(name: String, priority: Priority) extends Completion
 
+  /**
+    * Represents a keyword literal completion (i.e. `true`)
+    *
+    * @param name      the name of the keyword.
+    * @param priority  the priority of the keyword.
+    */
+  case class KeywordLiteralCompletion(name: String, priority: Priority) extends Completion
+
+  /**
+    * Represents a completion for a keyword with call syntax (i.e. `checked_cast`)
+    * 
+    * @param name      the name of the keyword.
+    * @param priority  the priority of the keyword.
+    */
+  case class KeywordCallCompletion(name: String, priority: Priority) extends Completion
+
+  /**
+    * Represents a collection keyword completion (i.e. `Hash#{}`)
+    *
+    * @param name      the name of the keyword.
+    * @param priority  the priority of the keyword.
+    */
+  case class KeywordCollectionCompletion(name: String, priority: Priority) extends Completion
```

Comment:
Are these really keywords or should we put this support elsewhere?

Reply:
Array's require a region. How do you deal with that? This seems to suggest a snippet.

Reply:
Remove, cant work well.

Reply:
I know we came to the conclusion that these couldn't work well, but wouldn't it work fine if they were just suggested above their respective modules? Like, if  `List#{...}` is simply ranked above suggestions to module functions like `List.map`?